cardano-api/src/Cardano/Api/ProtocolParametersUpdate.hs

```diff
@@ -0,0 +1,674 @@
+       protocolUpdateMaxTxExUnits :: Maybe MaxTxExecutionUnits,
+
+       -- | Max total script execution resources units allowed per block
+       protocolUpdateMaxBlockExUnits :: Maybe MaxBlockExecutionUnits,
+
+       -- | Max size of a Value in a tx output.
+       protocolUpdateParamMaxValSize :: Maybe Natural
+    }
+  deriving (Eq, Show)
+
+instance Semigroup ProtocolParametersUpdate where
+    ppu1 <> ppu2 =
+      ProtocolParametersUpdate {
+        protocolUpdateProtocolVersion     = merge protocolUpdateProtocolVersion
+      , protocolUpdateDecentralization    = merge protocolUpdateDecentralization
+      , protocolUpdateExtraPraosEntropy   = merge protocolUpdateExtraPraosEntropy
+      , protocolUpdateMaxBlockHeaderSize  = merge protocolUpdateMaxBlockHeaderSize
+      , protocolUpdateMaxBlockBodySize    = merge protocolUpdateMaxBlockBodySize
+      , protocolUpdateMaxTxSize           = merge protocolUpdateMaxTxSize
+      , protocolUpdateTxFeeFixed          = merge protocolUpdateTxFeeFixed
+      , protocolUpdateTxFeePerByte        = merge protocolUpdateTxFeePerByte
+      , protocolUpdateMinUTxOValue        = merge protocolUpdateMinUTxOValue
+      , protocolUpdateStakeAddressDeposit = merge protocolUpdateStakeAddressDeposit
+      , protocolUpdateStakePoolDeposit    = merge protocolUpdateStakePoolDeposit
+      , protocolUpdateMinPoolCost         = merge protocolUpdateMinPoolCost
+      , protocolUpdatePoolRetireMaxEpoch  = merge protocolUpdatePoolRetireMaxEpoch
+      , protocolUpdateStakePoolTargetNum  = merge protocolUpdateStakePoolTargetNum
+      , protocolUpdateMonetaryExpansion   = merge protocolUpdateMonetaryExpansion
+      , protocolUpdatePoolPledgeInfluence = merge protocolUpdatePoolPledgeInfluence
+      , protocolUpdateTreasuryCut         = merge protocolUpdateTreasuryCut
+      -- Intoduced in Alonzo below.
+      , protocolUpdateUTxOCostPerByte     = merge protocolUpdateUTxOCostPerByte
+      , protocolUpdateCostModels          = mergeMap protocolUpdateCostModels
+      , protocolUpdatePrices              = mergeMap protocolUpdatePrices
+      , protocolUpdateMaxTxExUnits        = merge protocolUpdateMaxTxExUnits
+      , protocolUpdateMaxBlockExUnits     = merge protocolUpdateMaxBlockExUnits
+      , protocolUpdateParamMaxValSize     = merge protocolUpdateParamMaxValSize
+      }
+      where
+        -- prefer the right hand side:
+        merge :: (ProtocolParametersUpdate -> Maybe a) -> Maybe a
+        merge f = f ppu2 `mplus` f ppu1
+
+        -- need to use map merge where:
+        -- keys present in m1 and m2: remove/ignore m1 key
+        -- keys present in m1 and not m2 : remove/ignore m1 key
+        -- keys present in m2 and not m1: keep m2
+        -- I.e prefer m2 -- TODO: left off here. Remember plutus json jared sent you
+        mergeMap :: (ProtocolParametersUpdate -> Map AnyScriptLanguage a)
+                 -> Map AnyScriptLanguage a
+        mergeMap = undefined
```

Comment:
I don't follow this one:
```         -- keys present in m1 and not m2 : remove/ignore m1 key
```
Why?

I _think_ we just want a simple `Map.union` but flipped, so that we prefer the second arg, e.g.
```
mergeMap f = f ppu2 `Map.union` f ppu1
```
But that does contradict your comment:
```         -- keys present in m1 and not m2 : remove/ignore m1 key
```
since we'd keep the keys present in m1, since they're not overridden by m2.